src/mc.rs

```diff
@@ -1003,10 +1005,13 @@ unsafe fn resize_rust<BD: BitDepth>(
     for y in 0..h {
         let mut mx = mx0;
         let mut src_x = -1 - 3;
-        let dst = dst.offset(y as isize * BD::pxstride(dst_stride));
+        let dst = dst + (y as isize * dst.pixel_stride::<BD>());
         let src = src + (y as isize * src.pixel_stride::<BD>());
         let src = &*src.slice::<BD>(src_w);
-        let dst = slice::from_raw_parts_mut(dst, dst_w);
+        let dst = match dst.data {
+            PicOrBuf::Pic(pic) => &mut *pic.slice_mut::<BD, _>((dst.offset.., ..dst_w)),
+            PicOrBuf::Buf(buf) => &mut *buf.mut_slice_as((dst.offset.., ..dst_w)),
+        };
```

Comment:
No, because this uses lifetime extension and the guard types are not the same, only the lifetime extended slices from dereferencing the guards.  Lifetime extension doesn't work through functions, so only a macro would work here, and I don't think that's worth it (Maybe it could? What do you think?).  I tried a while to get this kind of thing to work until I realized that I'm pretty sure it's impossible.  I guess a closure could also work, but that's unwieldy.

Reply:
Nah, I didn't catch that this was doing lifetime extension and that we had different guard types. So doing the match locally makes sense, I don't think a macro is worth it here